controlplane/kubeadm/controllers/helpers.go

```diff
@@ -100,6 +100,7 @@ func (r *KubeadmControlPlaneReconciler) reconcileExternalReference(ctx context.C
 
 	// Note: We intentionally do not handle checking for the paused label on an external template reference
 
+	// NOTE: This object doesn't implement observedGeneration. No need to add the WithStatusObservedGeneration option.
```

Comment:
This comment seems a bit misleading, we shouldn't be updating observedGeneration because we are not managing this resource, just trying to get information from an externally reconciled resource.